functions/dataManger/gameManager.ts

```diff
@@ -0,0 +1,176 @@
+          ":sortKey": "[GameItem]"
+      }
+    };
+    
+    try {
+      let response = await docClient.query(params).promise();      
+      let gameList = [] as any
+      response.Items.forEach((game: Interfaces.IDynamoObject) => {
+        let returnedGame = serializeDynamoResponse(game);
+        gameList.push(returnedGame);
+      });
+      return gameList;
+    } catch (err: any) {
+      throw err
+    }
+  }
+  export async function modifyGame(game: Game) { 
+    let template = await generateModifyExpression(game);
+    let params = {
+      TableName: table,
+      Key: {
+        partitionKey: game.partitionKey,
+        sortKey: game.sortKey
+      },
+      KeyConditionExpression: `sortKey = ${game.sortKey} AND partitionKey = ${game.partitionKey}`,
+      UpdateExpression: `SET ${template.updateExpression.join(",")}`,
+      ExpressionAttributeNames: template.expressionAttributeNames,
+      ExpressionAttributeValues: template.expressionAttributeValues,
+      ConditionExpression: 'attribute_exists(partitionKey) and attribute_exists(sortKey)',
+      ReturnValues: 'ALL_NEW'
+    };
+  
+    try {
+      let response = await docClient.update(params).promise();
+      let modifiedGame = await getGame(game);
+      return modifiedGame;
+    } catch (err: any) {
+      if (err.message == "The conditional request failed") {
+        throw new GameError("Unable to modify game.", 400);
+      }
+      throw err;
+    }
+  }
+  
+  export async function deleteGame(game: Game) {
+    let params = {
+      TableName: table,
+      Key: {
+        partitionKey: game.partitionKey,
+        sortKey: game.sortKey
+      },
+      KeyConditionExpression: `sortKey = ${game.sortKey} AND partitionKey = ${game.partitionKey}`,
+      ConditionExpression: 'attribute_exists(sortKey) and attribute_exists(partitionKey)',
+      ReturnValues: 'ALL_OLD'
+    };
+  
+    try {
+      let response = await docClient.delete(params).promise();
+      let game = serializeDynamoResponse(response.Attributes);
+      return game;      
+    } catch (err: any) {
+      if (err.message == "The conditional request failed") {
+        throw new GameError("Unable to delete game.", 400);
+      }
+      throw err;
+    }
+  }
+
+  export async function generateModifyExpression(game: Game) : Promise<Interfaces.IUpdateExpression>{
+    let updateExpression: String[] = [];
+    let expressionAttributeNames = {} as any;
+    let expressionAttributeValues = {} as any;
+    
+    //Generate dynammic update expression based on allowed parameters
+    for (let [key, value] of Object.entries(game)) {
+      if (key != 'partitionKey' && key != 'gameName' && key != 'sortKey' && value != undefined) {
+        updateExpression.push(`#${key} = :${key}`);
+        expressionAttributeNames[`#${key}`] = key ;
+        expressionAttributeValues[`:${key}`] = value;  
+      }
+    }
+    //Whenever a game is modified, this rechecks the price
+    if (game.desiredPrice) {
```

Comment:
We should discuss the architecture here. No problems but some interesting ideas to consider.